diff -ur hdf5-1.12.0.orig/src/H5Zdeflate.c hdf5-1.12.0/src/H5Zdeflate.c
--- hdf5-1.12.0.orig/src/H5Zdeflate.c	2020-02-28 17:29:58.000000000 -0600
+++ hdf5-1.12.0/src/H5Zdeflate.c	2025-01-03 13:40:44.638407469 -0600
@@ -91,7 +91,7 @@
     if (flags & H5Z_FLAG_REVERSE) {
 	/* Input; uncompress */
 	z_stream	z_strm;                 /* zlib parameters */
-	size_t		nalloc = *buf_size;     /* Number of bytes for output (compressed) buffer */
+	size_t		nalloc = *buf_size * 2; /* Number of bytes for output (compressed) buffer */
 
         /* Allocate space for the compressed buffer */
 	if (NULL==(outbuf = H5MM_malloc(nalloc)))
