diff -ru hdf5-1.8.5-patch1.orig/test/dt_arith.c hdf5-1.8.5-patch1/test/dt_arith.c
--- hdf5-1.8.5-patch1.orig/test/dt_arith.c	2010-08-26 09:21:40.000000000 -0500
+++ hdf5-1.8.5-patch1/test/dt_arith.c	2010-09-18 16:08:11.000000000 -0500
@@ -5177,8 +5177,8 @@
 static int
 run_fp_int_conv(const char *name)
 {
-#ifdef H5_FP_TO_INTEGER_OVERFLOW_WORKS
     int		nerrors = 0;
+#ifdef H5_FP_TO_INTEGER_OVERFLOW_WORKS
     int         test_values;
 
 #ifdef H5_VMS
